### PR TITLE
add github action for reserved keywords linter (PSRE-12)

### PR DESCRIPTION
**Pull request by nadeemshahzad**
**Description:**

The purpose of this PR is to add keywords linter to Django ida in cookiecutters repo.

**JIRA:**

https://openedx.atlassian.net/browse/PSRE-12

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -38,3 +38,11 @@ jobs:
...
         TOXENV: ${{ matrix.toxenv }}
       run: tox
+    - name: Run check-keywords
+      run: |
+        pip install -r requirements/base.txt
+        cookiecutter --no-input cookiecutter-django-ida
+        cd repo_name
+        make upgrade
+        pip install -r requirements/production.txt
+        make check_keywords
```
Comment (adzuci) [inline .github/workflows/ci.yml:48]: Just to confirm, does make check_keywords already return a non-zero exit code if it finds keywords / will this block PR merges if it finds a keyword that shouldn't be there?

**Review by adzuci — APPROVED** [review]
LGTM, I'm also adding @estute just as an FYI since I _think_ this will help reduce rework relating to data pipelines and keywords.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -38,3 +38,11 @@ jobs:
       env:
         TOXENV: ${{ matrix.toxenv }}
       run: tox
+    - name: Run check-keywords
```
Comment (adzuci) [inline .github/workflows/ci.yml:41]: I wonder if we should add a comment mentioning why this is being added and what to do if you add a keyword and see an error?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (nadeemshahzad) [reply .github/workflows/ci.yml:48]: Yes, this will fail the check and action and will block the merge.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (nadeemshahzad) [reply .github/workflows/ci.yml:41]: thanks, @adzuci  I've added a comment here.

**Pull request merged**